diff --git a/lib/src/bindings/remote.dart b/lib/src/bindings/remote.dart
--- a/lib/src/bindings/remote.dart
+++ b/lib/src/bindings/remote.dart
@@ -1,7 +1,7 @@
 import 'dart:ffi';
 import 'dart:ffi' as ffi;

-import 'package:ffi/ffi.dart' show Arena, calloc, using;
+import 'package:ffi/ffi.dart' show Arena, using;
 import 'package:git2dart/src/bindings/remote_callbacks.dart';
 import 'package:git2dart/src/callbacks.dart';
 import 'package:git2dart/src/extensions.dart';
@@ -479,9 +479,9 @@ void fetch({
   String? proxyOption,
 }) {
   using((arena) {
-    final refspecsC = calloc<git_strarray>();
+    final refspecsC = arena<git_strarray>();
     final refspecsPointers = refspecs.map((e) => e.toChar(arena)).toList();
-    final strArray = calloc<Pointer<Char>>(refspecs.length);
+    final strArray = arena<Pointer<Char>>(refspecs.length);

     for (var i = 0; i < refspecs.length; i++) {
       strArray[i] = refspecsPointers[i];
@@ -493,7 +493,7 @@ void fetch({

     final proxyOptions = _proxyOptionsInit(proxyOption, arena);

-    final opts = calloc<git_fetch_options>();
+    final opts = arena<git_fetch_options>();
     libgit2.git_fetch_options_init(opts, GIT_FETCH_OPTIONS_VERSION);

     opts.ref.pruneAsInt = prune;

---
